src/inject.ts

```diff
@@ -0,0 +1,112 @@
+interface Coordinate {
+  x: number;
+  y: number;
+}
+
+(function () {
+  let doNotRerenderTopshare = false;
+
+  function addConvoy(
+      centerOfConvoyingUnit: Coordinate,
+      centerOfConvoyedUnit: Coordinate,
+      centerOfConvoyDestination: Coordinate,
+      didConvoyFail: boolean
+  ): void {
+    let yInflectionIncrement: number;
+    let xInflectionIncrement: number;
+    const midpointXCoordinate = (centerOfConvoyedUnit.x + centerOfConvoyDestination.x) / 2,
```

Comment:
Much clearer now, yay! But it looks rather verbose - it might be helpful to think about if a simplifying abstraction could be used.